### ISSUE TITLE
Problem with enqueueReadBuffer

### ISSUE THREAD

**Issue by roastedpork** [HPCE/hpce-2017-cw3#43]
I am currently working on `StepWorldV3OpenCL` according to the readme, and my code simply throws an `Exception : clEnqueueReadBuffer` message. My system also gave a `/usr/local/cuda/lib64/libOpenCL.so.1: no version information available` warning message, but otherwise the programme ran fine until the `enqueueReadBuffer` portion was added.

How exactly does the command queue fail when reading from the buffer, and is there any way to fix this?

**Comment by roastedpork** [comment]
Adding onto this issue, I have tried placing the function within a try block as such:
```cpp
cl_int res = 1; // To check if the result has truly been modified, as all error values are negative
try{
	res = queue.enqueueReadBuffer(buffBuffer, CL_TRUE, 0, cbBuffer, &buffer[0], &copyBackDependencies);
	std::cerr << "cl_int value: " << res << std::endl;
}catch(...){
	std::cerr << "Read failure" << std::endl;
}
```
In some cases I obtained a `cl-int value: 0`, which corresponds to `CL_SUCCESS` under the documentation. However, the print out of the state contained values such as `-nan` or `-9086479269384671575069434354597888.00000000` which are obviously incorrect.

In other cases, I would simply obtain `Read failure`. This occurs whenever I change the world dimensions to a non-default size (e.g. 5 or 10), and/or after I ran the programme multiple times.
